src/server/mod.rs

```diff
@@ -1,20 +1,37 @@
 use anyhow::Error;
 use hyper::service::{make_service_fn, service_fn};
+use std::net::SocketAddr;
 
+use crate::config::tls::TlsConfig;
 use crate::config::Config;
 
 mod handler;
+mod https;
 mod service;
 
 pub struct Server {
     config: Config,
 }
 
 impl Server {
-    pub async fn serve(&self) {
+    pub async fn new(config: Config) -> Server {
```

Comment:
Unless you plan to add async operations here in the future, this shouldn't be async.

Reply:
Oh! Totally! This was part of a refactor and I left that by mistake!